py/build.gradle

```diff
@@ -111,8 +107,8 @@ task distPython(type: Exec, dependsOn: checkPythonEnv) {
         }
 
         copy {
-            from "${configurations.compile.join(',')}"
```

Comment:
So last update: not even this is possible, if we specify the project as dependency the configuration contains complete source set ( before when just shadow configuration was specified as a dependency, it contained just output of that task thus only desired jar file.)

We can't use the the jars provided by this source set since it wouldn't contain changes introduced by shadow jar such as the relocation of classes
So i don't know of any better solution without coupling the projects again. By accessing the configuration of different project we couple the projects. So this seems like the simplest and single functional solution I know of